docs: update njs build instructions

Jcahilltorre opened the pull request:
Update the `Enabling njs` steps to include changing branches to the latest supported version

![image](https://github.com/nginx/unit-docs/assets/78599298/341790d9-c838-49fd-ad5b-3f3163f01e05)

ac000 commented:
njs has [moved](https://mailman.nginx.org/pipermail/nginx/2024-June/WBTVAEMFIOQDL3VEBP7Z65Q4E56T73L6.html) to [GitHub](https://github.com/nginx/njs)

callahad commented on source/howto/source.rst:
```diff
@@ -115,23 +115,27 @@ revision numbers, respectively); omit the packages you won't use.
    .. code-block:: console
 
       $ cd ..
       $ git clone https://github.com/nginx/njs
+      $ git checkout 0.8.2
```
```suggestion
      $ git clone --branch 0.8.2 https://github.com/nginx/njs
```

or

```suggestion
      $ git clone https://github.com/nginx/njs
      $ git switch 0.8.2
```

ac000 commented on source/howto/source.rst:
```diff
@@ -115,23 +115,27 @@ revision numbers, respectively); omit the packages you won't use.
+   **0.8.2** is the latest version of :program:`njs` that Unit supports.
+   Make sure to check out the correct branch before configuring the binaries.
+
    If you'd like to use `Mercurial <https://www.mercurial-scm.org/downloads>`_:
 
    .. code-block:: console
 
       $ cd ..
       $ hg clone https://hg.nginx.org/njs
+      $ hg update -r 0.8.2
```
Seeing as njs has officially moved to GH I'd remove any mention of mercurial.

Not that it matters if you remove this, but it looks like there's a missing `cd njs` in there...

callahad replied:
Good catch on the `cd njs`; updated my suggestions below

callahad commented on source/howto/source.rst:
```diff
@@ -115,23 +115,27 @@ revision numbers, respectively); omit the packages you won't use.
    .. code-block:: console
 
       $ cd ..
       $ hg clone https://hg.nginx.org/njs
+      $ hg update -r 0.8.2
```
```suggestion
      $ hg clone --branch 0.8.2 https://hg.nginx.org/njs
```

or

```suggestion
      $ hg clone --update 0.8.2 https://hg.nginx.org/njs
```

(In Mercurial `--branch` _only_ clones that branch, `--update` clones everything and then checks out that branch.)

callahad replied:
...or just nuke mentions of mercurial altogether, per discussion in this issue.

callahad reviewed (approved):
Looks reasonable. Two small suggestions at your option.

callahad commented:
Seconding @ac000's suggestion of removing mentions of mercurial given the mailing list announcement he pointed to.

ac000 commented on source/howto/source.rst:
```diff
@@ -115,23 +115,27 @@ revision numbers, respectively); omit the packages you won't use.
 
    .. code-block:: console
 
       $ cd ..
       $ git clone https://github.com/nginx/njs
+      $ git checkout 0.8.2
```
Missing a `cd njs` and maybe better not putting people in a detached head state, not that it really matters just for building it but people may think somethings wrong...

```
$ git clone https://github.com/nginx/njs.git
$ cd njs
$ git checkout -b 0.8.2 0.8.2
```

callahad replied:
`git switch 0.8.2` would also take care of that

ac000 replied:
Yes, if they fully read the message...

You could also just go for

```
$ git clone --depth=1 --branch 0.8.2 https://github.com/nginx/njs.git
```

which will save a lot of downloading...

Jcahilltorre commented:
removed the mercurial steps
changed the git steps to 

```
 $ git clone https://github.com/nginx/njs.git
 $ cd njs
 $ git checkout -b 0.8.2 0.8.2
```

lcrilly commented on source/howto/source.rst:
```diff
@@ -115,29 +115,22 @@ revision numbers, respectively); omit the packages you won't use.
    .. code-block:: console
 
-      $ cd njs
       $ ./configure :nxt_hint:`--no-zlib --no-libxml2 <Ensures Unit can link against the resulting library>` && make
```
We should have a stronger, explicit hint that these configure arguments are essential.

Jcahilltorre replied:
Updated the text on line 130

ac000 commented on source/howto/source.rst:
```diff
@@ -115,29 +115,22 @@ revision numbers, respectively); omit the packages you won't use.
-
-      $ cd ..
-      $ git clone https://github.com/nginx/njs
+      $ git clone https://github.com/nginx/njs.git
```
If you do that, you may as well also use `--depth=1`

Jcahilltorre replied:
I didn’t take this path because doing it this way, I still get a warning about a detached HEAD.

I thought we wanted to avoid having customers in a detached HEAD state

lcrilly replied:
The warning is a bit alarming. Makes sense to avoid it.

Jcahilltorre commented on source/howto/source.rst:
```diff
@@ -115,29 +115,24 @@ revision numbers, respectively); omit the packages you won't use.
+      $ git checkout -b 0.8.2 0.8.2
 
-   Next, configure and build the :program:`njs` binaries:
+   Next, configure and build the :program:`njs` binaries. Make sure to use the
+   `--no-zlib` and `--no-libxml2` options to avoid
+   conflicts with Unit's dependencies:
```
is this better @lcrilly ?

lcrilly replied:
Perfect!
